xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -484,15 +484,26 @@ bool CRendererDXVA::CRenderBufferImpl::UploadToTexture()
   return m_bLoaded;
 }
 
-ProcessorConversion CRendererDXVA::ChooseConversion(const ProcessorConversions& conversions) const
+ProcessorConversion CRendererDXVA::ChooseConversion(
+    const ProcessorConversions& conversions, const SupportedConversionsArgs& conversionArgs) const
 {
   assert(conversions.size() > 0);
 
+  // The AMD dxva processor has low chroma upsampling quality (nearest neighbor) from YCbCr SDR
+  // colorspaces to 10 bit RGB, at least up to driver 31.0.21905.1001 / Adrenalin 23.11.1.
+  // As a workaround, fall back to 8 bit RGB output.
+
+  const bool force8bit = DX::DeviceResources::Get()->GetAdapterDesc().VendorId == PCIV_AMD &&
+                         !(conversionArgs.m_colorPrimaries == AVCOL_PRI_BT2020 &&
```

Comment:
This excludes HDR sources but what happens with 10 bits SDR? (x264 / H.264 10-bit). 

Should accurate chroma or preserve 10-bit precision take priority? is not clear to me.

Reply:
On purpose, the only thing that triggers the bug is SDR converted to 10 bit RGB. HDR to 10 bit is fine.

For SDR, I'm not aware of real content with 10 actual bits of precision, they're at least 99% artificial 10bit for improved encoding efficiency purposes. The chroma artifacts are much more noticeable to me than a difference of value of 1 on a pixel sometimes.